Source/Plugins/Core/com.equella.core/js/tsrc/modules/SearchModule.ts

```diff
@@ -202,3 +216,84 @@ export const searchItems = ({
   };
   return OEQ.Search.search(API_BASE_URL, searchParams);
 };
+
+/**
+ * A function that takes query string params from a shared searching.do url and converts all applicable params to Search options
+ * @param queryString query string params from a shared searching.do url
+ * @return SearchOptions object
+ */
+export const convertParamsToSearchOptions = async (queryString: string) => {
+  if (!queryString) return undefined;
+
+  const params = new URLSearchParams(queryString);
+
+  const getQueryParam = (paramName: string) => {
+    return params.get(paramName) ?? undefined;
+  };
+
+  const collection = getQueryParam(LegacySearchParams.IN)?.substring(1);
+  const owner = getQueryParam(LegacySearchParams.OWNER);
+  const dateRange = getQueryParam(LegacySearchParams.DATERANGE);
+
+  const getUserDetails = async (
+    userId: string
+  ): Promise<OEQ.UserQuery.UserDetails> => {
+    const userDetails = await resolveUsers([userId]);
+    return userDetails[0];
+  };
+
+  const getCollectionDetails = async (
+    collectionUuid: string
+  ): Promise<Collection[]> => {
+    const collectionList = await collectionListSummary(["SEARCH_COLLECTION"]);
+    return collectionList.filter((c) => {
+      return c.uuid === collectionUuid;
+    });
+  };
+
+  const getLastModifiedDateRange = (
+    rangeType: string,
+    primaryDate: Date,
+    secondaryDate?: Date
+  ): DateRange => {
+    let dr: DateRange = {};
+    switch (rangeType?.toLocaleLowerCase()) {
+      case undefined:
+        break;
+      case "between":
+        dr = { start: primaryDate, end: secondaryDate };
+        break;
+      case "after":
+        dr = { start: primaryDate ?? undefined };
+        break;
+      case "before":
+        dr = { end: primaryDate };
+        break;
+      case "on":
+        dr = { start: primaryDate, end: primaryDate };
+        break;
+      default:
+        throw new TypeError("Invalid range");
+        break;
+    }
+    return dr;
+  };
+
+  const searchOptions: SearchOptions = {
+    ...defaultSearchOptions,
+    collections: collection
+      ? await getCollectionDetails(collection)
+      : undefined,
+    query: getQueryParam(LegacySearchParams.QUERY),
+    owner: owner ? await getUserDetails(owner) : undefined,
+    lastModifiedDateRange: getLastModifiedDateRange(
+      dateRange ?? "",
+      new Date(parseInt(getQueryParam(LegacySearchParams.PRIMARYDATE) ?? "")),
+      new Date(parseInt(getQueryParam(LegacySearchParams.SECONDARYDATE) ?? ""))
+    ),
+    sortOrder:
+      (getQueryParam(LegacySearchParams.SORT)?.toUpperCase() as SortOrder) ||
```

Comment:
Should this be:

```suggestion
      (getQueryParam(LegacySearchParams.SORT)?.toUpperCase() as SortOrder) ??
```

?